src/main/scala/viper/gobra/translator/implementations/translator/PermissionsImpl.scala

```diff
@@ -0,0 +1,34 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.implementations.translator
+
+import viper.gobra.ast.internal.{FractionalPerm, FullPerm, NoPerm, WildcardPerm}
+import viper.gobra.ast.{internal => in}
+import viper.gobra.translator.interfaces.{Collector, Context}
+import viper.gobra.translator.interfaces.translator.Permissions
+import viper.gobra.translator.util.ViperWriter.CodeWriter
+import viper.silver.{ast => vpr}
+
+class PermissionsImpl extends Permissions {
```

Comment:
Is there a reason why you added this as an translator? I would have just add this a typeEncoding because later we will have to change it to that either way (when we want to support perm as a regular type). I nothing is speaking against this, I would ask you to add the permission enocoding as an type encoding, e.g. look at bool encoding.

Reply:
It seemed more natural to me and I think I got inspired by expressions / assertions. But I don't mind changing it to a type encoding